### DISCUSSION TITLE
Tensorflow tf not installed when trying to train classifier AI

### DISCUSSION THREAD

**Discussion by Kiesmith** [bitcraze/discussions#559] [Q&A]
I installed everything on pip3 through the requirements.txt in the classifier example. **Most of this is just logs and will be quick to go through as I have left a brief comment after each telling what it shows so you do not have to scan through everything. Making it fairly quick to go through.**

Note: I setup docker and the autotiler through the guide on "getting started with the aideck".

What I am trying to do: Run the train_classifier.py in classification example. 

Here is the log to prove it: 

**bitcraze@ubuntu:/aideck-gap8-examples/examples/ai/classification$ pip3 install -r requirements.txt**
Collecting absl-py==1.0.0
  Downloading absl_py-1.0.0-py3-none-any.whl (126 kB)
     |################################| 126 kB 2.8 MB/s 
Collecting astunparse==1.6.3
  Downloading astunparse-1.6.3-py2.py3-none-any.whl (12 kB)
Collecting cachetools==4.2.4
  Downloading cachetools-4.2.4-py3-none-any.whl (10 kB)
Collecting certifi==2021.10.8
  Downloading certifi-2021.10.8-py2.py3-none-any.whl (149 kB)
     |################################| 149 kB 3.8 MB/s 
Collecting charset-normalizer==2.0.9
  Downloading charset_normalizer-2.0.9-py3-none-any.whl (39 kB)
Collecting docopt==0.6.2
  Downloading docopt-0.6.2.tar.gz (25 kB)
Collecting flatbuffers==2.0
  Downloading flatbuffers-2.0-py2.py3-none-any.whl (26 kB)
Collecting gast==0.4.0
  Downloading gast-0.4.0-py3-none-any.whl (9.8 kB)
Collecting google-auth==2.3.3
  Downloading google_auth-2.3.3-py2.py3-none-any.whl (155 kB)
     |################################| 155 kB 3.9 MB/s 
Collecting google-auth-oauthlib==0.4.6
  Downloading google_auth_oauthlib-0.4.6-py2.py3-none-any.whl (18 kB)
Collecting google-pasta==0.2.0
  Downloading google_pasta-0.2.0-py3-none-any.whl (57 kB)
     |################################| 57 kB 1.6 MB/s 
Collecting grpcio==1.43.0
  Downloading grpcio-1.43.0-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.1 MB)
     |################################| 4.1 MB 7.0 MB/s 
Collecting h5py==3.6.0
  Downloading h5py-3.6.0-cp38-cp38-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (4.5 MB)
     |################################| 4.5 MB 4.9 MB/s 
Collecting idna==3.3
  Downloading idna-3.3-py3-none-any.whl (61 kB)
     |################################| 61 kB 4.3 MB/s 
Collecting importlib-metadata==4.10.0
  Downloading importlib_metadata-4.10.0-py3-none-any.whl (17 kB)
Collecting keras==2.7.0
  Downloading keras-2.7.0-py2.py3-none-any.whl (1.3 MB)
     |################################| 1.3 MB 5.2 MB/s 
Collecting Keras-Preprocessing==1.1.2
  Downloading Keras_Preprocessing-1.1.2-py2.py3-none-any.whl (42 kB)
     |################################| 42 kB 874 kB/s 
Collecting libclang==12.0.0
  Downloading libclang-12.0.0-2-py2.py3-none-manylinux1_x86_64.whl (13.3 MB)
     |################################| 13.3 MB 193 kB/s 
Collecting Markdown==3.3.6
  Downloading Markdown-3.3.6-py3-none-any.whl (97 kB)
     |################################| 97 kB 2.8 MB/s 
Collecting numpy==1.21.5
  Downloading numpy-1.21.5-cp38-cp38-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (15.7 MB)
     |################################| 15.7 MB 1.9 MB/s 
Collecting oauthlib==3.1.1
  Downloading oauthlib-3.1.1-py2.py3-none-any.whl (146 kB)
     |################################| 146 kB 2.5 MB/s 
Collecting opt-einsum==3.3.0
  Downloading opt_einsum-3.3.0-py3-none-any.whl (65 kB)
     |################################| 65 kB 100 kB/s 
Collecting Pillow==8.4.0
  Downloading Pillow-8.4.0-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (3.1 MB)
     |################################| 3.1 MB 4.8 MB/s 
Collecting pipreqs==0.4.11
  Downloading pipreqs-0.4.11-py2.py3-none-any.whl (32 kB)
Collecting protobuf==3.19.1
  Downloading protobuf-3.19.1-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.1 MB)
     |################################| 1.1 MB 5.1 MB/s 
Collecting pyasn1==0.4.8
  Downloading pyasn1-0.4.8-py2.py3-none-any.whl (77 kB)
     |################################| 77 kB 3.2 MB/s 
Collecting pyasn1-modules==0.2.8
  Downloading pyasn1_modules-0.2.8-py2.py3-none-any.whl (155 kB)
     |################################| 155 kB 4.7 MB/s 
Collecting requests==2.26.0
  Downloading requests-2.26.0-py2.py3-none-any.whl (62 kB)
     |################################| 62 kB 42 kB/s 
Collecting requests-oauthlib==1.3.0
  Downloading requests_oauthlib-1.3.0-py2.py3-none-any.whl (23 kB)
Collecting rsa==4.8
  Downloading rsa-4.8-py3-none-any.whl (39 kB)
Collecting scipy==1.7.3
  Downloading scipy-1.7.3-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (39.3 MB)
     |################################| 39.3 MB 25 kB/s 
Collecting six==1.16.0
  Downloading six-1.16.0-py2.py3-none-any.whl (11 kB)
Collecting tensorboard==2.7.0
  Downloading tensorboard-2.7.0-py3-none-any.whl (5.8 MB)
     |################################| 5.8 MB 3.8 MB/s 
Collecting tensorboard-data-server==0.6.1
  Downloading tensorboard_data_server-0.6.1-py3-none-manylinux2010_x86_64.whl (4.9 MB)
     |################################| 4.9 MB 48 kB/s 
Collecting tensorboard-plugin-wit==1.8.0
  Downloading tensorboard_plugin_wit-1.8.0-py3-none-any.whl (781 kB)
     |################################| 781 kB 36 kB/s 
Collecting tensorflow==2.8.0
  Downloading tensorflow-2.8.0-cp38-cp38-manylinux2010_x86_64.whl (497.6 MB)
     |################################| 497.6 MB 297 bytes/s 
Collecting tensorflow-estimator==2.7.0
  Downloading tensorflow_estimator-2.7.0-py2.py3-none-any.whl (463 kB)
     |################################| 463 kB 6.0 MB/s 
Collecting tensorflow-io-gcs-filesystem==0.23.1
  Downloading tensorflow_io_gcs_filesystem-0.23.1-cp38-cp38-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (2.1 MB)
     |################################| 2.1 MB 1.6 MB/s 
Collecting termcolor==1.1.0
  Downloading termcolor-1.1.0.tar.gz (3.9 kB)
Collecting typing_extensions==4.0.1
  Downloading typing_extensions-4.0.1-py3-none-any.whl (22 kB)
Collecting urllib3==1.26.7
  Downloading urllib3-1.26.7-py2.py3-none-any.whl (138 kB)
     |################################| 138 kB 2.6 MB/s 
Collecting Werkzeug==2.0.2
  Downloading Werkzeug-2.0.2-py3-none-any.whl (288 kB)
     |################################| 288 kB 4.7 MB/s 
Collecting wrapt==1.13.3
  Downloading wrapt-1.13.3-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl (84 kB)
     |################################| 84 kB 591 kB/s 
Collecting yarg==0.1.9
  Downloading yarg-0.1.9-py2.py3-none-any.whl (19 kB)
Collecting zipp==3.6.0
  Downloading zipp-3.6.0-py3-none-any.whl (5.3 kB)
ERROR: Could not find a version that satisfies the requirement scipi==1.22.3 (from -r requirements.txt (line 46)) (from versions: 0.1, 1.0, 2.0, 3.0)
ERROR: No matching distribution found for scipi==1.22.3 (from -r requirements.txt (line 46))
bitcraze@ubuntu:~/aideck-gap8-examples/examples/ai/classification$ pip3 install scipi==2.0
Collecting scipi==2.0
  Downloading scipi-2.0-py3-none-any.whl (12 kB)
Installing collected packages: scipi
Successfully installed scipi-2.0

**I received an error about scipi so I changed it to 2.0.** 
**I check the list to confirm:** 

**bitcraze@ubuntu:~/aideck-gap8-examples/examples/ai/classification$ pip3 list**
Package                Version              Location                                            
---------------------- -------------------- ----------------------------------------------------
appdirs                1.4.4                
asyncqt                0.8.0                
blinker                1.4                  
certifi                2019.11.28           
cfclient               2022.3rc2            /home/bitcraze/projects/crazyflie-clients-python/src
cflib                  0.1.18.1             /home/bitcraze/projects/crazyflie-lib-python        
chardet                3.0.4                
Click                  7.0                  
colorama               0.4.3                
command-not-found      0.3                  
cryptography           2.8                  
dbus-python            1.2.16               
distro                 1.4.0                
distro-info            0.23ubuntu1          
entrypoints            0.3                  
freetype-py            2.2.0                
hsluv                  5.0.2                
httplib2               0.14.0               
idna                   2.8                  
keyring                18.0.1               
kiwisolver             1.4.0                
language-selector      0.1                  
launchpadlib           1.10.13              
lazr.restfulclient     0.14.2               
lazr.uri               1.0.3                
libusb-package         1.0.25.0             
netifaces              0.10.4               
numpy                  1.22.3               
oauthlib               3.1.0                
olefile                0.46                 
opencv-python-headless 4.5.5.64             
pexpect                4.6.0                
Pillow                 7.0.0                
pip                    20.0.2               
Pygments               2.3.1                
PyGObject              3.36.0               
PyJWT                  1.7.1                
pymacaroons            0.13.0               
PyNaCl                 1.3.0                
PyQt5                  5.15.6               
PyQt5-Qt5              5.15.2               
PyQt5-sip              12.9.1               
pyqtgraph              0.12.4               
pyserial               3.5                  
python-apt             2.0.0+ubuntu0.20.4.7 
python-dateutil        2.7.3                
pyusb                  1.2.1                
PyYAML                 5.3.1                
pyzmq                  22.3.0               
qtm                    2.1.1                
reportlab              3.5.34               
requests               2.22.0               
requests-unixsocket    0.2.0                
scipi                  2.0                  
scipy                  1.8.0                
SecretStorage          2.3.1                
setuptools             45.2.0               
simplejson             3.16.0               
sip                    4.19.21              
six                    1.14.0               
ssh-import-id          5.10                 
systemd-python         234                  
ubuntu-advantage-tools 27.6                 
ubuntu-drivers-common  0.0.0                
ufw                    0.36                 
unattended-upgrades    0.1                  
urllib3                1.25.8               
vispy                  0.9.6                
wadllib                1.3.3                
wheel                  0.34.2               
wxPython               4.0.7                
xkit                   0.0.0                
bitcraze@ubuntu:~/aideck-gap8-examples/examples/ai/classification$ pip3 show tensorflow
WARNING: Package(s) not found: tensorflow

**Tensorflow not seen, so I install it manually:**

**bitcraze@ubuntu:~/aideck-gap8-examples/examples/ai/classification$ pip3 install tensorflow==2.8.0**
Collecting tensorflow==2.8.0
  Using cached tensorflow-2.8.0-cp38-cp38-manylinux2010_x86_64.whl (497.6 MB)
Collecting h5py>=2.9.0
  Downloading h5py-3.8.0-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.7 MB)
     |################################| 4.7 MB 142 kB/s 
Collecting tf-estimator-nightly==2.8.0.dev2021122109
  Downloading tf_estimator_nightly-2.8.0.dev2021122109-py2.py3-none-any.whl (462 kB)
     |################################| 462 kB 705 kB/s 
Collecting typing-extensions>=3.6.6
  Downloading typing_extensions-4.4.0-py3-none-any.whl (26 kB)
Requirement already satisfied: setuptools in /usr/lib/python3/dist-packages (from tensorflow==2.8.0) (45.2.0)
Collecting tensorboard<2.9,>=2.8
  Downloading tensorboard-2.8.0-py3-none-any.whl (5.8 MB)
     |################################| 5.8 MB 1.9 MB/s 
Collecting gast>=0.2.1
  Downloading gast-0.5.3-py3-none-any.whl (19 kB)
Collecting grpcio<2.0,>=1.24.3
  Downloading grpcio-1.51.1-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (4.8 MB)
     |################################| 4.8 MB 819 kB/s 
Collecting libclang>=9.0.1
  Downloading libclang-15.0.6.1-py2.py3-none-manylinux2010_x86_64.whl (21.5 MB)
     |################################| 21.5 MB 66 kB/s 
Collecting termcolor>=1.1.0
  Downloading termcolor-2.2.0-py3-none-any.whl (6.6 kB)
Collecting protobuf>=3.9.2
  Downloading protobuf-4.21.12-cp37-abi3-manylinux2014_x86_64.whl (409 kB)
     |################################| 409 kB 1.7 MB/s 
Requirement already satisfied: six>=1.12.0 in /usr/lib/python3/dist-packages (from tensorflow==2.8.0) (1.14.0)
Collecting astunparse>=1.6.0
  Using cached astunparse-1.6.3-py2.py3-none-any.whl (12 kB)
Collecting wrapt>=1.11.0
  Downloading wrapt-1.14.1-cp38-cp38-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (81 kB)
     |################################| 81 kB 1.6 MB/s 
Collecting tensorflow-io-gcs-filesystem>=0.23.1
  Downloading tensorflow_io_gcs_filesystem-0.30.0-cp38-cp38-manylinux_2_12_x86_64.manylinux2010_x86_64.whl (2.4 MB)
     |################################| 2.4 MB 2.4 MB/s 
Collecting absl-py>=0.4.0
  Downloading absl_py-1.4.0-py3-none-any.whl (126 kB)
     |################################| 126 kB 1.0 MB/s 
Collecting google-pasta>=0.1.1
  Using cached google_pasta-0.2.0-py3-none-any.whl (57 kB)
Collecting flatbuffers>=1.12
  Downloading flatbuffers-23.1.21-py2.py3-none-any.whl (26 kB)
Requirement already satisfied: numpy>=1.20 in /home/bitcraze/.local/lib/python3.8/site-packages (from tensorflow==2.8.0) (1.22.3)
Collecting keras<2.9,>=2.8.0rc0
  Downloading keras-2.8.0-py2.py3-none-any.whl (1.4 MB)
     |################################| 1.4 MB 2.1 MB/s 
Collecting opt-einsum>=2.3.2
  Using cached opt_einsum-3.3.0-py3-none-any.whl (65 kB)
Collecting keras-preprocessing>=1.1.1
  Using cached Keras_Preprocessing-1.1.2-py2.py3-none-any.whl (42 kB)
Requirement already satisfied: requests<3,>=2.21.0 in /usr/lib/python3/dist-packages (from tensorboard<2.9,>=2.8->tensorflow==2.8.0) (2.22.0)
Collecting werkzeug>=0.11.15
  Downloading Werkzeug-2.2.2-py3-none-any.whl (232 kB)
     |################################| 232 kB 2.2 MB/s 
Requirement already satisfied: wheel>=0.26 in /usr/lib/python3/dist-packages (from tensorboard<2.9,>=2.8->tensorflow==2.8.0) (0.34.2)
Collecting google-auth-oauthlib<0.5,>=0.4.1
  Using cached google_auth_oauthlib-0.4.6-py2.py3-none-any.whl (18 kB)
Collecting markdown>=2.6.8
  Downloading Markdown-3.4.1-py3-none-any.whl (93 kB)
     |################################| 93 kB 479 kB/s 
Collecting tensorboard-plugin-wit>=1.6.0
  Downloading tensorboard_plugin_wit-1.8.1-py3-none-any.whl (781 kB)
     |################################| 781 kB 1.8 MB/s 
Collecting google-auth<3,>=1.6.3
  Downloading google_auth-2.16.0-py2.py3-none-any.whl (177 kB)
     |################################| 177 kB 2.1 MB/s 
Collecting tensorboard-data-server<0.7.0,>=0.6.0
  Using cached tensorboard_data_server-0.6.1-py3-none-manylinux2010_x86_64.whl (4.9 MB)
Collecting MarkupSafe>=2.1.1
  Downloading MarkupSafe-2.1.2-cp38-cp38-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (25 kB)
Collecting requests-oauthlib>=0.7.0
  Downloading requests_oauthlib-1.3.1-py2.py3-none-any.whl (23 kB)
Collecting importlib-metadata>=4.4; python_version < "3.10"
  Downloading importlib_metadata-6.0.0-py3-none-any.whl (21 kB)
Collecting cachetools<6.0,>=2.0.0
  Downloading cachetools-5.3.0-py3-none-any.whl (9.3 kB)
Collecting pyasn1-modules>=0.2.1
  Using cached pyasn1_modules-0.2.8-py2.py3-none-any.whl (155 kB)
Collecting rsa<5,>=3.1.4; python_version >= "3.6"
  Downloading rsa-4.9-py3-none-any.whl (34 kB)
Requirement already satisfied: oauthlib>=3.0.0 in /usr/lib/python3/dist-packages (from requests-oauthlib>=0.7.0->google-auth-oauthlib<0.5,>=0.4.1->tensorboard<2.9,>=2.8->tensorflow==2.8.0) (3.1.0)
Collecting zipp>=0.5
  Downloading zipp-3.13.0-py3-none-any.whl (6.7 kB)
Collecting pyasn1<0.5.0,>=0.4.6
  Using cached pyasn1-0.4.8-py2.py3-none-any.whl (77 kB)
Installing collected packages: h5py, tf-estimator-nightly, typing-extensions, MarkupSafe, werkzeug, requests-oauthlib, cachetools, pyasn1, pyasn1-modules, rsa, google-auth, google-auth-oauthlib, zipp, importlib-metadata, markdown, absl-py, tensorboard-plugin-wit, protobuf, grpcio, tensorboard-data-server, tensorboard, gast, libclang, termcolor, astunparse, wrapt, tensorflow-io-gcs-filesystem, google-pasta, flatbuffers, keras, opt-einsum, keras-preprocessing, tensorflow
ERROR: Could not install packages due to an EnvironmentError: [Errno 28] No space left on device: '/tmp/pip-unpacked-wheel-sh4qjs03/tensorflow/python/_pywrap_tensorflow_internal.so' -> '/home/bitcraze/.local/lib/python3.8/site-packages/tensorflow/python/_pywrap_tensorflow_internal.so'
 
**My VM is new and it started it with 40gb, so I do not know why it says "no space left" when there is clearly allot, but it seems to have installed tensorflow**

**I check list again and it shows tensorflow:**

**bitcraze@ubuntu:~/aideck-gap8-examples/examples/ai/classification$ pip3 list**
Package                      Version              Location                                            
---------------------------- -------------------- ----------------------------------------------------
absl-py                      1.4.0                
appdirs                      1.4.4                
astunparse                   1.6.3                
asyncqt                      0.8.0                
blinker                      1.4                  
cachetools                   5.3.0                
certifi                      2019.11.28           
cfclient                     2022.3rc2            /home/bitcraze/projects/crazyflie-clients-python/src
cflib                        0.1.18.1             /home/bitcraze/projects/crazyflie-lib-python        
chardet                      3.0.4                
Click                        7.0                  
colorama                     0.4.3                
command-not-found            0.3                  
cryptography                 2.8                  
dbus-python                  1.2.16               
distro                       1.4.0                
distro-info                  0.23ubuntu1          
entrypoints                  0.3                  
flatbuffers                  23.1.21              
freetype-py                  2.2.0                
gast                         0.5.3                
google-auth                  2.16.0               
google-auth-oauthlib         0.4.6                
google-pasta                 0.2.0                
grpcio                       1.51.1               
h5py                         3.8.0                
hsluv                        5.0.2                
httplib2                     0.14.0               
idna                         2.8                  
importlib-metadata           6.0.0                
keras                        2.8.0                
Keras-Preprocessing          1.1.2                
keyring                      18.0.1               
kiwisolver                   1.4.0                
language-selector            0.1                  
launchpadlib                 1.10.13              
lazr.restfulclient           0.14.2               
lazr.uri                     1.0.3                
libclang                     15.0.6.1             
libusb-package               1.0.25.0             
Markdown                     3.4.1                
MarkupSafe                   2.1.2                
netifaces                    0.10.4               
numpy                        1.22.3               
oauthlib                     3.1.0                
olefile                      0.46                 
opencv-python-headless       4.5.5.64             
opt-einsum                   3.3.0                
pexpect                      4.6.0                
Pillow                       7.0.0                
pip                          20.0.2               
protobuf                     4.21.12              
pyasn1                       0.4.8                
pyasn1-modules               0.2.8                
Pygments                     2.3.1                
PyGObject                    3.36.0               
PyJWT                        1.7.1                
pymacaroons                  0.13.0               
PyNaCl                       1.3.0                
PyQt5                        5.15.6               
PyQt5-Qt5                    5.15.2               
PyQt5-sip                    12.9.1               
pyqtgraph                    0.12.4               
pyserial                     3.5                  
python-apt                   2.0.0+ubuntu0.20.4.7 
python-dateutil              2.7.3                
pyusb                        1.2.1                
PyYAML                       5.3.1                
pyzmq                        22.3.0               
qtm                          2.1.1                
reportlab                    3.5.34               
requests                     2.22.0               
requests-oauthlib            1.3.1                
requests-unixsocket          0.2.0                
rsa                          4.9                  
scipi                        2.0                  
scipy                        1.8.0                
SecretStorage                2.3.1                
setuptools                   45.2.0               
simplejson                   3.16.0               
sip                          4.19.21              
six                          1.14.0               
ssh-import-id                5.10                 
systemd-python               234                  
tensorboard                  2.8.0                
tensorboard-data-server      0.6.1                
tensorboard-plugin-wit       1.8.1                
**tensorflow                   2.8.0**                
**tensorflow-io-gcs-filesystem 0.30.0**               
termcolor                    2.2.0                
tf-estimator-nightly         2.8.0.dev2021122109  
typing-extensions            4.4.0                
ubuntu-advantage-tools       27.6                 
ubuntu-drivers-common        0.0.0                
ufw                          0.36                 
unattended-upgrades          0.1                  
urllib3                      1.25.8               
vispy                        0.9.6                
wadllib                      1.3.3                
Werkzeug                     2.2.2                
wheel                        0.34.2               
wrapt                        1.14.1               
wxPython                     4.0.7                
xkit                         0.0.0                
zipp                         3.13.0               
bitcraze@ubuntu:~/aideck-gap8-examples/examples/ai/classification$ pip3 show tensorlow
WARNING: Package(s) not found: tensorlow

**bitcraze@ubuntu:~/aideck-gap8-examples/examples/ai/classification$ pip3 show tensorflow**
Name: tensorflow
Version: 2.8.0
Summary: TensorFlow is an open source machine learning framework for everyone.
Home-page: https://www.tensorflow.org/
Author: Google Inc.
Author-email: packages@tensorflow.org
License: Apache 2.0
Location: /home/bitcraze/.local/lib/python3.8/site-packages
Requires: astunparse, numpy, six, keras, tensorflow-io-gcs-filesystem, grpcio, libclang, google-pasta, tf-estimator-nightly, protobuf, gast, keras-preprocessing, tensorboard, wrapt, setuptools, opt-einsum, absl-py, h5py, termcolor, typing-extensions, flatbuffers
Required-by: 

**I decide to run the trainer.py since I have tensorflow installed:**

**bitcraze@ubuntu:~/aideck-gap8-examples/examples/ai/classification$ python3 train_classifier.py [--args]**
Traceback (most recent call last):
  File "train_classifier.py", line 27, in <module>
    import tensorflow as tf
  File "/home/bitcraze/.local/lib/python3.8/site-packages/tensorflow/__init__.py", line 37, in <module>
    from tensorflow.python.tools import module_util as _module_util
ModuleNotFoundError: No module named 'tensorflow.python.tools'

**I have been having this issue for a long time now and I am finding it difficult to get by. I also tried doing this in a miniconda environment and it gave the exact same response.**

Extra: About the disk error, it says I have more than enough space when I check the vm
![image](https://user-images.githubusercontent.com/72498028/218337382-cec4f41c-33e1-4ef2-8fba-a9f668eb684a.png)

Update:
I made another VM and I installed the latest tensorflow version, got rid of the old tensorflow stuff in the requirements.txt and run it. I tried the trainer.py again and it said  showed another issue saying: 
"The Tensorflow library was compiled to use the AVX instructions, but these aren't available on your machine. Aborted (core dumped)."

**Comment by hmllr** [comment]
Hi :)

There are indeed some problems in the requirements list, I just updated it in my ongoing pull request, so you can already check out the requirements I just tested in a clean virtual environment here: https://github.com/bitcraze/aideck-gap8-examples/pull/111/commits/7e063582c76fdd5fc556de1f60df6e6bef07f382

However, I think your final problem is still tensorflow related and not really Crazyflie/AIdeck related, as it seems as if your CPU doesn't support AVX and pip installs a version of tensorflow which requires it. This is reported in the tensorflow repo, for example here: https://github.com/tensorflow/tensorflow/issues/24548 

Hope this helps,
Hanna

**Reply by Kiesmith** [reply]
I managed to train the AI on miniconda with the latest TensorFlow. I just tested this AI and after flashing it onto the deck and I realised that the AI I tested is not working.

I only changed the dimensions 224x224 (in the trainer.py code). It it formatted correctly in the training dataset, and it does return things in the console window and say "object detected", but it only seems to constantly say that when it looks at anything. the dataset I used to train the AI is this: https://www.kaggle.com/datasets/constantinwerner/human-detection-dataset

I understand that the code converts images to greyscale. 

**Any advice on this or do u know the solution?** 
Also: I have a pretrained model that I made myself that does actually work on my own camera. I have been trying to fit that one on instead. I have converted it to a tensorflow lite model but I havent quantized it. 

**Is it possible to only use a non-quantized version? I also get errors when I try to quantasize my none-related model (make it into a "_q.tflight" type).**

Thanks,
Kieran

**Reply by hmllr** [reply]
Hi :)

Your question is not related to the title anymore - please post a new question (otherwise others might not be able to re-use the answers as it will be hard to find).
